src/main/python/covata/delta/crypto/cryptoservice.py

```
@@ -0,0 +1,160 @@
+
+import base64
+import os
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from binascii import hexlify
+
+from ..util import LogMixin
+from .signer import CVTSigner
+
+
+class CryptoService(LogMixin):
+    def __init__(self,
+                 key_store_path,
+                 key_store_passphrase):
+        # type: (str, str) -> self
+        """
+        Constructs a new Crypto service with the given configuration.
+
+        :param str key_store_path: the path to the private key store
+        :param str key_store_passphrase: the passphrase to decrypt the keys
+        """
+        self.key_store_path = os.path.expanduser(key_store_path)
+        self.__key_store_passphrase = key_store_passphrase
+
+    def save(self, private_key, file_name):
+        """
+        Saves a private key object (encrypted) to keystore
+
+        Saving the Private Cryptographic Key:
+
+        >>> crypto_service.save(private_key, identity_id + ".crypto.pem")
+
+        Saving the Private Signing Key:
+
+        >>> crypto_service.save(private_key, identity_id + ".signing.pem")
+
+
+        :param private_key: the private key object
+        :type private_key: :class:`RSAPrivateKey`
+        :param str file_name: the name of the .pem file to be written
+        """
+        # type: (rsa.RSAPrivateKey, str) -> None
+        pem = private_key.private_bytes(
+            serialization.Encoding.PEM,
+            serialization.PrivateFormat.PKCS8,
+            serialization.BestAvailableEncryption(self.__key_store_passphrase))
+
+        file_path = os.path.join(self.key_store_path, file_name)
+        if not os.path.isdir(self.key_store_path):
+            self.logger.debug("creating directory %s", self.key_store_path)
+            os.makedirs(self.key_store_path)
+
+        with open(file_path, 'w') as f:
```

sure, what is the desired behaviour when a duplicate already exists? should we raise an exception?